File: packages/ik_rules/lib/src/world/map_label.dart
import 'package:ik_content/ik_content.dart';

/// Map IDs that must not show this location, from [LocationRow.hiddenOnMapIDs].
List<String> hiddenMapIdsFor(LocationRow location) {
  final raw = location.hiddenOnMapIDs;
  if (raw == null || raw.trim().isEmpty) return const <String>[];
  return raw
      .split(';')
      .map((part) => part.trim())
      .where((part) => part.isNotEmpty)
      .toList(growable: false);
}

bool locationHiddenOnMap(LocationRow location, String mapId) {
  return hiddenMapIdsFor(location).contains(mapId);
}

/// Gateway nodes use [LocationRow.mapNodeName] on a child sub-map.
String mapNodeLabel(LocationRow location, String browseMapId) {
  final mapId = location.mapId;
  final nodeName = location.mapNodeName;
  if (nodeName != null && nodeName.isNotEmpty && mapId != null && browseMapId != mapId) {
    return nodeName;
  }
  return location.displayName;
}

File: src/game/world/mapLabel.ts
import type { LocationRow } from '../data/types'

/** Map IDs that must not show this location, from Hidden On Map IDs. */
export function hiddenMapIdsFor(location: LocationRow): string[] {
  const raw = location['Hidden On Map IDs']
  if (typeof raw !== 'string' || !raw.trim()) return []
  return raw
    .split(';')
    .map((part) => part.trim())
    .filter((part) => part.length > 0)
}

export function locationHiddenOnMap(location: LocationRow, mapId: string): boolean {
  return hiddenMapIdsFor(location).includes(mapId)
}

/** Gateway nodes use Map Node Name on a child sub-map. */
export function mapNodeLabel(location: LocationRow, browseMapId: string): string {
  const mapId = location['Map ID']
  const nodeName = location['Map Node Name']
  if (nodeName && mapId && browseMapId !== mapId) return nodeName
  return location['Display Name']
}
